WordPress/gutenberg
# Move Global Styles code to `lib/compat/wordpress-5.9` folder

Related https://github.com/WordPress/gutenberg/pull/36957#issuecomment-981759580
~Depends on https://github.com/WordPress/gutenberg/pull/36974~

This PR moves the global styles classes code under `lib/compat/wordpress-5.9` folder. The rationale for this is to make sure we have a path to remove this code from the plugin.

Files moved to the 5.9 directory:

- `lib/class-wp-theme-json-gutenberg.php`
- `lib/class-wp-theme-json-resolver-gutenberg.php`
- `lib/class-wp-theme-json-schema-gutenberg.php`
- `lib/theme.json`
- `lib/theme-i18n.json`

## How to test

- Verify that automated tests pass.
- Use a theme you're familiar with (I did it with TT1-blocks) and verify that styles look as expected in the post editor, site editor, and front end.
- Also check that the presets (colors, font sizes) are present in the block inspector and global styles sidebar.
